### PR TITLE
Add Creative Commons Catalog API

### PR DESCRIPTION
**Pull request by krysal**
Hi! Great collection of links here! I just wanted to drop an additional API that can be of help.

Currently CC Catalog only serves images, but it's planned to add additional media types such as open texts and audio, with the ultimate goal of providing access to all 1.4 billion CC licensed and public domain works on the web. See the [web portal](https://search.creativecommons.org/).

So it can be categorized under **Photography** or **Open Data** (maybe), but I put it in **Open Source Projects** for now.

The API supports `OAuth2` but is not required. There are limitations for anonymous users though.

Awaiting for comments about this entry.

## Checklist

- [x] Your submissions are formatted according to the guidelines in the [contributing guide](CONTRIBUTING.md)
- [x] Your additions are ordered alphabetically
- [x] Your submission has a useful description
- [x] The description does not end with punctuation
- [x] Each table column should be padded with one space on either side
- [x] You have searched the repository for any relevant issues or pull requests
- [x] Any category you are creating has the minimum requirement of 3 items
- [x] All changes have been [squashed][squash-link] into a single commit

[squash-link]: <https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit>

### CODE REVIEW

**Review by 53jk1 — APPROVED** [review]

**Comment by krysal** [comment]
@yannbertrand Please point out if you think anything is needed to merge this.

Thanks!

File: README.md [hunk 1]
```diff
@@ -617,6 +617,7 @@ API | Description | Auth | HTTPS | CORS |
 API | Description | Auth | HTTPS | CORS |
 |---|---|---|---|---|
 | [Countly](https://api.count.ly/reference) | Countly web analytics | No | No | Unknown |
+| [Creative Commons Catalog](https://api.creativecommons.engineering/) | Search among openly licensed and public domain works | No | Yes | Yes |
```
Comment (yannbertrand) [inline README.md:620]:
```suggestion
| [Creative Commons Catalog](https://api.creativecommons.engineering/) | Search among openly licensed and public domain works | `OAuth` | Yes | Yes |
```

Comment (yannbertrand) [reply README.md:620]: I think most of the endpoints are only available through OAuth

Comment (krysal) [reply README.md:620]: There is the option to use the API without an API key too, it is not strictly required :) but it's certainly more limited than the authenticated via.

Comment (yannbertrand) [reply README.md]: I see, I misread the introduction. Thanks for the explanations!

**Comment by yannbertrand** [comment]
Thank you!

**Pull request merged**